additional_data: codelist_code: Add some more codelist data

michaelwood opened the pull request:
Adds some of the more obscure codelists for easier human reading:
* locationScope
* beneficiary geoCodeType
* recipient_organization geoCodeType
* funding_organization geoCodeType

Related: https://github.com/ThreeSixtyGiving/grantnav/issues/1081
Related: https://github.com/ThreeSixtyGiving/standard/issues/391

Commit:
additional_data: codelist_code: Add some more codelist data

Adds some of the more obscure codelists for easier human reading:
* locationScope
* beneficiary geoCodeType
* recipient_organization geoCodeType
* funding_organization geoCodeType

Related: https://github.com/ThreeSixtyGiving/grantnav/issues/1081

R2ZER0 commented on datastore/additional_data/sources/codelist_code.py:
```diff
@@ -83,11 +98,52 @@ def update_additional_data(self, grant, additional_data):
+        try:
+            code = grant["locationScope"]
+            locationScope = CodelistCode.objects.get(
+                code=code, list_name="locationScope"
```
Minor suggestion: I guess we could add constants for `CodelistCode.list_name`  https://github.com/ThreeSixtyGiving/datastore/blob/live/datastore/additional_data/models.py#L167 in the same way we have for `OrgInfoCache.org_type` https://github.com/ThreeSixtyGiving/datastore/blob/live/datastore/additional_data/models.py#L13-L31 ?

michaelwood replied:
True that would be neater. We would need to refactor to avoid there being two sources for the `list_name` string as the import code looks it up from the code list file name https://github.com/ThreeSixtyGiving/datastore/blob/live/datastore/additional_data/sources/codelist_code.py#L29

will have another look

michaelwood replied:
I think I'd prefer to leave this optimisation as it requires refactoring old code that isn't currently in scope for this small bit of work.

R2ZER0 commented on datastore/additional_data/sources/codelist_code.py:
```diff
@@ -83,11 +98,52 @@ def update_additional_data(self, grant, additional_data):
+            pass
+
+        try:
+            for location in grant["beneficiaryLocation"]:
```
Should the try block be inside the for loop here, otherwise if the first location can't be found, it won't even try the rest?

R2ZER0 replied:
'cause of `CodelistCode.objects.get` raising

michaelwood replied:
yep good point

Commit:
additional_data: codelist: Remove changes needed to support geoCodeType

This codelist is currently not able to be supported.
See https://github.com/ThreeSixtyGiving/standard/issues/391

Backout the work done for this with the aim of creating a draft PR that
could revert this commit.

R2ZER0 reviewed (approved):
Looks good :+1:
